src/silo/storage/dictionary.cpp

```diff
@@ -56,302 +95,110 @@ void Dictionary::updateDictionary(
-      return error_string;
-   }
-   return pango_lineage_lookup.at(pango_lineage_id_in_lookup);
+std::string Dictionary::getPangoLineage(uint32_t pango_lineage_id_in_lookup) const {
+   return reverse_pango_lineage_lookup.at("pango_lineage").at(pango_lineage_id_in_lookup);
 }
 
 uint32_t Dictionary::getCountryIdInLookup(const std::string& country) const {
-   if (country_dictionary.contains(country)) {
-      return country_dictionary.at(country);
-   }
-   return UINT32_MAX;
+   return lookupValueId("country", country).value_or(UINT32_MAX);
 }
 
-const std::string& Dictionary::getCountry(uint32_t country_id_in_lookup) const {
-   if (country_id_in_lookup == UINT32_MAX) {
-      return error_string;
-   }
-   return country_lookup.at(country_id_in_lookup);
+std::string Dictionary::getCountry(uint32_t country_id_in_lookup) const {
+   return lookupStringValue("country", country_id_in_lookup).value_or("NOID");
 }
 
 uint32_t Dictionary::getRegionIdInLookup(const std::string& region) const {
-   if (region_dictionary.contains(region)) {
-      return region_dictionary.at(region);
-   }
-   return UINT32_MAX;
+   return lookupValueId("region", region).value_or(UINT32_MAX);
 }
 
-const std::string& Dictionary::getRegion(uint32_t region_lookup_id) const {
-   if (region_lookup_id == UINT32_MAX) {
-      return error_string;
-   }
-   return region_lookup.at(region_lookup_id);
+std::string Dictionary::getRegion(uint32_t region_lookup_id) const {
+   return lookupStringValue("region", region_lookup_id).value_or("NOID");
 }
 
-uint64_t Dictionary::getIdInGeneralLookup(const std::string& region_id_in_lookup) const {
-   if (general_dictionary.contains(region_id_in_lookup)) {
-      return general_dictionary.at(region_id_in_lookup);
-   }
-   return UINT64_MAX;
+uint32_t Dictionary::getPangoLineageCount() const {
+   return pango_lineage_lookup.at("pango_lineage").size();
+}
+uint32_t Dictionary::getCountryCount() const {
+   return string_lookup.at("country").size();
+}
+uint32_t Dictionary::getRegionCount() const {
+   return string_lookup.at("region").size();
+}
+// NOLINTNEXTLINE(readability-convert-member-functions-to-static)
+uint64_t Dictionary::getIdInGeneralLookup(const std::string& /*region_id_in_lookup*/) const {
+   return UINT32_MAX;
+}
+// NOLINTNEXTLINE(readability-convert-member-functions-to-static)
+std::string Dictionary::getGeneralLookup(uint64_t /*general_id_in_lookup*/) const {
+   return "";
+}
+// NOLINTNEXTLINE(readability-convert-member-functions-to-static)
+uint32_t Dictionary::getColumnIdInLookup(const std::string& /*column_name*/) const {
+   return UINT32_MAX;
+}
+// NOLINTNEXTLINE(readability-convert-member-functions-to-static)
+std::string Dictionary::getColumn(uint32_t /*column_id_in_lookup*/) const {
+   return "";
 }
 
-[[maybe_unused]] const std::string& Dictionary::getGeneralLookup(uint64_t general_id_in_lookup
+std::optional<ValueId> Dictionary::lookupValueId(
+   const std::string& column_name,
```

Comment:
Let's also use the alias here
```suggestion
   const ColumnName& column_name,
```